framework/src/utils/MooseEnumItem.C

```diff
@@ -0,0 +1,68 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+// MOOSE includes
+#include "MooseEnumItem.h"
+#include "MooseUtils.h"
+
+MooseEnumItem::MooseEnumItem(const std::string & name, const int & id)
+  : _raw_name(MooseUtils::trim(name)),
+    _name(MooseUtils::toUpper(_raw_name)),
+    _id(id),
+    _unique(std::hash<std::string>()(std::to_string(_id) + _name))
```

Comment:
I am capturing a discussion here:

This was done to protect when two MooseEnumItem objects are compared from different Moose(Multi)Enum objects, so that "foo=1" != "bar=1".

The solution is to not allow this and throw a exception in the operator==(MooseEnumInfo) overload.